src/Symfony/Component/Form/Extension/DataCollector/FormDataExtractor.php

```diff
@@ -24,16 +28,27 @@
 class FormDataExtractor implements FormDataExtractorInterface
 {
     /**
-     * @var ValueExporter
+     * @var VarCloner
      */
-    private $valueExporter;
+    private $varCloner;
 
+    private $varExporter;
     /**
      * Constructs a new data extractor.
+     *
+     * @param VarCloner|null $varCloner Passing a ValueExporter instance is 
+     *                                  deprecated since version 3.2 and will 
+     *                                  not be possible in 4.0.
      */
-    public function __construct(ValueExporter $valueExporter = null)
+    public function __construct($varCloner = null)
```

Comment:
to make this BC, I think we should still use the valueexporter by default and when injected,
and switch to VarCloner when such an instance is injected.
The modify `form_debug.xml` to inject a var cloner